File: api_nestjs_distributed_order_platform/src/application/port/event-publisher.repository.ts
import { OrderEntity } from "src/domain/entities/order.entity";
import { PaymentEntity } from "src/domain/entities/payment.entity";
import { StockEntity } from "src/domain/entities/stock.entity";


export interface EventPublisherRepositoryInterface {
    publishCreatedOrder(order: OrderEntity): Promise<void>;
    publishConfirmedOrder(order: OrderEntity): Promise<void>;
    publishCancelledOrder(order: OrderEntity): Promise<void>;
    publishPaymentPaid(payment: PaymentEntity): Promise<void>;
    publishPaymentFailed(payment: PaymentEntity): Promise<void>;
    publishStockReserved(stock: StockEntity): Promise<void>;
    publishStockReleased(stock: StockEntity): Promise<void>;
    publishStockConsumed(stock: StockEntity): Promise<void>;
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/port/EventPublisherRepository.java
package com.project.order.application.port;

import com.project.order.domain.model.Order;
import com.project.order.domain.model.Payment;
import com.project.order.domain.model.Stock;

public interface EventPublisherRepository {
    void publishCreatedOrder(Order order);
    void publishConfirmedOrder(Order order);
    void publishCancelledOrder(Order order);
    void publishPaymentPaid(Payment payment);
    void publishPaymentFailed(Payment payment);
    void publishStockReserved(Stock stock);
    void publishStockReleased(Stock stock);
    void publishStockConsumed(Stock stock);
}
